packages/components/src/lib/controllers/app-controller.js

```diff
@@ -892,18 +892,22 @@ export default class AppController {
       homonym.lexemes.filter((l) => l.isPopulated()).length < 1
     if (notFound && !this._store.state.app.queryStillActive) {
       let languageName
+      let languageCode
       if (homonym) {
+        languageCode = homonym.language
         languageName = this.api.app.getLanguageName(homonym.languageID).name
-      } else if (this._store.state.app.currentLanguageName) {
+      } else if (this._store.state.app.currentLanguageCode && this._store.state.app.currentLanguageName) {
+        languageCode = this._store.state.app.currentLanguageCode
```

Comment:
If we do not have homonym, we do not have the `homonym.language` either. In that case we should take language code from the `currentLanguageCode` as we do with the language name.